### PR TITLE
Adds documentation index.

### PR DESCRIPTION
**Pull request by filmil**
The documentation index is a table of document names and their
respective first paragraphs.  This should help a bit in document
navigation, especially as the number of documents grows.

Also added a navigation link from README.md, as this is the likely
project entry point for at least a little while.

Fixes: #56

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,3 +19,7 @@ ICU4X, or "ICU for X", will be built from the start with several key design cons
 4. Written by i18n experts to encourage best practices.
 
 ICU4X will provide an ECMA-402-compatible API surface in the target client-side platforms, including the web platform, iOS, Android, WearOS, WatchOS, Flutter, and Fuchsia, supported in programming languages including Rust, JavaScript, Objective-C, Java, Dart, and C++.
+
+## Documentation
+
+The effort documentation can be found in [the docs subdirectory](docs/index.md).
```
Comment (nciric) [inline README.md:25]: More information about the project can be found in [the docs subdirectory](docs/index.md).

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Documentation
```
Comment (nciric) [inline docs/index.md:1]: I wonder, is there a auto tool for this?

Comment (sffc) [reply docs/index.md:1]: We should plan to push the docs directory to github.io.  I'm not sure whether you are supposed to use index.md or README.md when you do that.

File: README.md (hunk 1 above)
Comment (filmil) [reply README.md]: Done

File: docs/index.md (hunk 2 above)
Comment (filmil) [reply docs/index.md:1]: @nciric Not familiar, though what I did was rather mechanical: inserted the first paragraph of each document into a .md table.

@sffc sgtm; until that time, perhaps this works.

Any other actionable feedback?

**Pull request merged**